### ISSUE TITLE
Get variable error (using NULL?)

### ISSUE THREAD

**Issue by reinholdsson** [SwedishPensionsAgency/Coldbir#74]
This question might be twofold - first, I'm not sure if it's ok to still use `a["x", NULL]` - if not, we should handle that.

However, for the second part, the behaviour is really strange - seems somehow to be related with the use of `data.table`. Before running, please **restart the R-session**. Any ideas?

```

# Setup
library(testthat)
library(Coldbir)
a <- cdb(tempfile(), read_only = F)

# Set variable
a["x", NULL] <- sample(1:5, 1e3, replace = T)
a["x", NULL]

# Warning message:
#   In is.na(j) : is.na() applied to non-(list or vector) of type 'NULL'

# Set variable again
a["x", NULL] <- sample(1:5, 1e3, replace = T)
a["x", NULL]

# Warning message:
#   In is.na(j) : is.na() applied to non-(list or vector) of type 'NULL'

# Run tests
test()
# => warning and errors

# Set variable again
a["x", NULL] <- sample(1:5, 1e3, replace = T)
a["x", NULL]

# Error in `:=`(len, unlist(lapply(dims, FUN = length))) : 
#   := and `:=`(...) are defined for use in j, once only and in particular ways. See help(":="). Check is.data.table(DT) is TRUE.
# In addition: Warning message:
#   In is.na(j) : is.na() applied to non-(list or vector) of type 'NULL'
```

**Comment by mikula** [comment]
the problem with a["x", NULL] is easy. We could allow this kind of request in the current version. NULL had a meaning in the early stage of this issue.

**Comment by reinholdsson** [comment]
Running (also after `test()`):

```
a[] <- x
a[]
```

leads to about the same issue:

```
Browse[1]> toRead[,len:= unlist(lapply(dims, FUN= length))]  # ERROR?
Error in `:=`(len, unlist(lapply(dims, FUN = length))) : 
  := and `:=`(...) are defined for use in j, once only and in particular ways. See help(":="). Check is.data.table(DT) is TRUE.
```

The error is one the following line: https://github.com/SwedishPensionsAgency/Coldbir/blob/issue55/R/cdb-class.r#L718

**Comment by reinholdsson** [comment]
Changing

`toRead[,len:= unlist(lapply(dims, FUN= length))]`

to

`toRead$len <- unlist(lapply(toRead$dims, FUN= length))`

solves the first issue, but leads to another one:

`Error in paste(dims[[1]], collapse = ".") : object 'dims' not found`

Somehow, it might have a problem having `dims` as a column name.

**Comment by reinholdsson** [comment]
We will temporarly solve it by changing toRead to a list+vector instead of using a `data.table`.

**Comment by mikula** [comment]
Excellent. Keep it simple.

**Comment by reinholdsson** [comment]
Solved, see #76.
